diff --git a/Jolt/Physics/SoftBody/SoftBodyMotionProperties.cpp b/Jolt/Physics/SoftBody/SoftBodyMotionProperties.cpp
index 4aa88d2..a2d036f 100644
--- a/Jolt/Physics/SoftBody/SoftBodyMotionProperties.cpp
+++ b/Jolt/Physics/SoftBody/SoftBodyMotionProperties.cpp
@@ -104,6 +104,15 @@ void SoftBodyMotionProperties::Initialize(const SoftBodyCreationSettings &inSett
 	CalculateMassAndInertia();
 }
 
+void SoftBodyMotionProperties::RotateRodFrames(QuatArg inRotation)
+{
+	for (RodState &rod : mRodStates)
+	{
+		rod.mRotation = (inRotation * rod.mRotation).Normalized();
+		rod.mAngularVelocity = inRotation * rod.mAngularVelocity;
+	}
+}
+
 float SoftBodyMotionProperties::GetVolumeTimesSix() const
 {
 	float six_volume = 0.0f;
diff --git a/Jolt/Physics/SoftBody/SoftBodyMotionProperties.h b/Jolt/Physics/SoftBody/SoftBodyMotionProperties.h
index 8768efd..d78fada 100644
--- a/Jolt/Physics/SoftBody/SoftBodyMotionProperties.h
+++ b/Jolt/Physics/SoftBody/SoftBodyMotionProperties.h
@@ -63,6 +63,7 @@ public:
 	/// Access to the state of rods
 	Quat								GetRodRotation(uint inIndex) const			{ return mRodStates[inIndex].mRotation; }
 	Vec3								GetRodAngularVelocity(uint inIndex) const	{ return mRodStates[inIndex].mAngularVelocity; }
+	void								RotateRodFrames(QuatArg inRotation);
 
 	/// Get the materials of the soft body
 	const PhysicsMaterialList &			GetMaterials() const						{ return mSettings->mMaterials; }
